content/guides/end-to-end-testing/amazon-cognito-authentication.md

```diff
@@ -157,6 +158,169 @@ const awsConfig = require('./aws-exports-es5.js')
+  // eslint-disable-next-line cypress/no-unnecessary-waiting
+  cy.wait(2000)
+
+  // verify we have made it passed the login screen
+  cy.contains('Get Started').should('be.visible')
+})
+```
+
+Secondly, we can use our `loginByCognito` command in the test. Below is our test
+to login as a user via [Amazon Cognito](https://aws.amazon.com/cognito),
+complete the onboarding process and logout.
+
+<Alert type="success">
+
+The
+[runnable version of this test](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/tests/ui-auth-providers/cognito.spec.ts)
+is in the
+[Cypress Real World App](https://github.com/cypress-io/cypress-realworld-app).
+
+</Alert>
+
+```jsx
+describe('Cognito', function () {
+  beforeEach(function () {
+    // Seed database with test data
+    cy.task('db:seed')
+
+    // login via Amazon Cognito via cy.origin()
+    cy.loginByCognito(
+      Cypress.env('cognito_username'),
+      Cypress.env('cognito_password')
+    )
+  })
+
+  it('shows onboarding', function () {
+    cy.contains('Get Started').should('be.visible')
+  })
+})
+```
+
+<DocsVideo src="/img/examples/aws-cognito-origin.mp4"></DocsVideo>
+
+Lastly, we can refactor our login command to take advantage of
+[`cy.session()`](/api/commands/session) to store our logged in user so we don't
+have to reauthenticate with everything test.
+
+```jsx
+// Amazon Cognito
+Cypress.Commands.add('loginByCognito', (username, password) => {
+  cy.session(
+    `cognito-${username}`,
+    () => {
+      Cypress.log({
+        displayName: 'COGNITO LOGIN',
+        message: [`🔐 Authenticating | ${username}`],
+        // @ts-ignore
+        autoEnd: false,
+      })
+
+      cy.visit('/')
+      cy.contains('Sign in with AWS', {
+        includeShadowDom: true,
+      }).click()
+
+      cy.origin(
+        Cypress.env('cognito_domain'),
+        {
+          args: {
+            username,
+            password,
+          },
+        },
+        ({ username, password }) => {
+          // cognito log in page has some elements of the same id but are off screen. we only want the visible elements to log in
+          cy.get('input[name="username"]:visible').type(username)
+          cy.get('input[name="password"]:visible').type(password, {
+            log: false,
+          })
+          cy.get('input[name="signInSubmitButton"]:visible').click()
+        }
+      )
+
+      // give a few seconds for redirect to settle
+      // eslint-disable-next-line cypress/no-unnecessary-waiting
+      cy.wait(2000)
+
+      // verify we have made it passed the login screen
+      cy.contains('Get Started').should('be.visible')
+    },
+    {
+      validate() {
+        cy.visit('/')
+        // revalidate our session to make sure we are logged in
+        cy.contains('Get Started').should('be.visible')
+      },
+    }
+  )
+})
+```
+
+<DocsVideo src="/img/examples/cognito-session-restore.mp4"></DocsVideo>
+
+Unlike programmatic login, authenticating with
```

Comment:
ah maybe jumping the gun? I haven't ready about programatic login yet! Maybe move this to the section `Adapting an Amazon Cognito App for Testing`.

We could all out there, if you want to use cy.orgion, you can skip this section. and/or maybe we can update the title to say something like `Adapting an Amazon Cognito App for Programmatic Login`.

Reply:
I added a little info disclaimer in the section instead of changing the title in e14bf7170. How does this look?

<img width="690" alt="Screen Shot 2022-12-01 at 12 44 49 PM" src="https://user-images.githubusercontent.com/3980464/205123659-f89187bd-0063-41c2-9e9c-2eb32d01e32d.png">